src/devices/BuildHat/Sensors/Sensor.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using Iot.Device.BuildHat.Models;
+
+namespace Iot.Device.BuildHat.Sensors
+{
+    /// <summary>
+    /// Interface for a sensor
+    /// </summary>
+    public class Sensor
+    {
+        internal Brick Brick;
+
+        /// <summary>
+        /// Gets true if the motor is connected.
+        /// </summary>
+        public bool IsConnected { get; internal set; }
+
+        /// <summary>
+        /// Creates a sensor
+        /// </summary>
+        /// <param name="brick">The brick.</param>
+        /// <param name="port">The port.</param>
+        /// <param name="type">The sensor type.</param>
+        internal Sensor(Brick brick, SensorPort port, SensorType type)
+        {
+            Brick = brick;
+            Port = port;
+            SensorType = type;
+        }
+
+        /// <summary>
+        /// Gets the name of the sensor.
+        /// </summary>
+        /// <returns>The sensor name.</returns>
+        public virtual string SensorName { get => "Generic sensor"; }
+
+        /// <summary>
+        /// Gets the Sensor port
+        /// </summary>
+        /// <returns>The sensor port</returns>
+        public SensorPort Port { get; internal set; }
+
+        /// <summary>
+        /// Gets the sensor type
+        /// </summary>
+        public SensorType SensorType { get; internal set; }
+
+        /// <summary>
+        /// To notify a property has changed.
+        /// </summary>
+        public event PropertyChangedEventHandler? PropertyChanged;
+
+        /// <summary>
+        /// To notify a property has changed.
+        /// </summary>
+        public event PropertyChangedEventHandler? PropertyUpdated;
+
+        internal void OnPropertyChanged(string name) => PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
```

Comment:
Add `[CallerMemberName]` attribute.

Reply:
What will that bring in the current implementation? I may have missed something.

Reply:
If you declare your member as 

```
internal void OnPropertyChanged([CallerMemberName]string name=null)
```

You can call `OnPropertyChanged()` within a property without argument. The compiler will automatically insert the current property name at the call location. It's just a shortcut.

Reply:
I think I'll skip this. reason is: it's not supported in nanoFramework and I want to port it to nano. So if I adjust the shortcut here, I'll anyway will have to go for the long way.

Reply:
Oh, Ok then.